Sources/Basic/OutputByteStream.swift

```diff
@@ -55,32 +58,108 @@ public class OutputByteStream: OutputStream {
         return buffer.count
     }
 
-    /// The contents of the output stream.
-    ///
-    /// This method implicitly flushes the stream.
-    public var bytes: ByteString {
-        flush()
-        return ByteString(self.buffer)
+    /// Currently available buffer size.
+    private var availableBufferSize: Int {
+        return max(buffer.capacity, OutputByteStream.bufferSize) - buffer.count
     }
-    
+
+    /// Clears the buffer maintaining current capacity.
+    private func clearBuffer() {
+        buffer.removeAll(keepingCapacity: true)
+    }
+
     // MARK: Data Output API
 
     public func flush() {
-        // Do nothing.
+        writeImpl(ArraySlice(buffer))
+        buffer.removeAll(keepingCapacity: true)
     }
 
     /// Write an individual byte to the buffer.
     public func write(_ byte: UInt8) {
+        // If buffer is full, write and clear it.
+        if availableBufferSize == 0 {
+            writeImpl(ArraySlice(buffer))
+            clearBuffer()
+        }
```

Comment:
Pathological, but we should document the buffer size must always be non-zero somewhere (i.e., this would need to change if we supported unbuffered).